dvc/command/experiments.py

```diff
@@ -399,8 +409,13 @@ def show_experiments(
         include_params=include_params,
         exclude_params=exclude_params,
     )
-    metric_headers = _normalize_headers(metric_names)
-    param_headers = _normalize_headers(param_names)
+
+    names = {**metric_names, **param_names}
+    counter = Counter(
+        name for path in names for name in names[path] for path in names
```

Comment:
$ names = {"a":{"b":1, "c":2}}
$ [name for path in names for name in names[path] for path in names]
> ['b', 'c']

$ names = {"a":{"b":1, "c":2}, "d":{}}
$ [name for path in names for name in names[path] for path in names]
> ['b', 'b', 'c', 'c']

A bug ?

Reply:
I think the bug is just that there is an extra `for path in names` in the original expression for some reason.

But what this does is count the number of times a particular metric or param name is used. Since different metrics/params files can use the same names for values (like `pipeline1/params.yaml:foo` and `pipeline2/params.yaml:foo`) we count the number of times that `foo` is used, and if it is once we can just display it as `foo` in the table. If it is used more than once, we have to display the columns using the full `path:foo` header instead.